### PR TITLE
Add a clang static analyser check and fix issues

### PR DESCRIPTION
**Pull request by mdiluz**
This uses the clang static analyzer (tool name is `scan-build`) to spot issues that we may have in CI. I've also fixed up the couple it spotted.

Note: the `artifacts` entry isn't active[ without an AWS hosting instance](https://docs.travis-ci.com/user/uploading-artifacts/) annoyingly, but I've put it there for future proofing as it'd be nice to collect the full log file from scan-build if needed.

Locally, the user generally needs the `clang-tools` package installed to run the script.

The failure at 23415ac for example looks like this:
```
...
[11/31] Linking static target daemon/libdaemon-common.a.
[12/31] Compiling C object 'daemon/f77b12a@@gamemoded@exe/gamemode.c.o'.
[13/31] Compiling C object 'daemon/f77b12a@@gamemoded@exe/gamemode-env.c.o'.
../../../daemon/gamemode.c:385:7: warning: Potential leak of memory pointed to by 'executable'
        if (!config_get_client_whitelisted(self->config, executable)) {
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 warning generated.
[14/31] Compiling C object 'daemon/f77b12a@@gamemoded@exe/gamemode-ioprio.c.o'.
[15/31] Compiling C object 'daemon/f77b12a@@gamemoded@exe/gamemode-tests.c.o'.
...
[21/31] Linking static target subprojects/inih/libinih.a.
[22/31] Compiling C object 'daemon/f77b12a@@gamemoded@exe/daemon_config.c.o'.
../../../daemon/gamemode-gpu.c:86:3: warning: Potential leak of memory pointed to by 'new_info'
                LOG_ERROR("Found invalid vendor, will not apply optimisations!\n");
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../../daemon/logging.h:58:7: note: expanded from macro 'LOG_ERROR'
                if (get_use_syslog()) {                                                                    \
                    ^~~~~~~~~~~~~~
1 warning generated.
[23/31] Compiling C object 'daemon/f77b12a@@gamemoded@exe/dbus_messaging.c.o'.
...
[31/31] Linking target daemon/gamemoded.
scan-build: 2 bugs found.
scan-build: Run 'scan-view /home/travis/build/mdiluz/gamemode/build/meson-logs/scanbuild/2019-05-19-114922-7102-1' to examine bug reports.
```

### CODE REVIEW

File: daemon/gamemode.c [hunk 1]
```diff
@@ -591,6 +595,7 @@ static GameModeClient *game_mode_client_new(pid_t pid, char *executable)
 		return NULL;
 	}
 	*ret = c;
+	strncpy(ret->executable, executable, PATH_MAX);
```
Comment (gicmo) [inline daemon/gamemode.c]: `strncpy` does not ensure a terminating `\0` byte, in the theoretically case of `strlen (executable) >= PATH_MAX` the `ret->executable` would not be `\0`-terminated. So maybe `strncpy(ret->executable, executable, PATH_MAX - 1);` to be sure?

Comment (mdiluz) [reply daemon/gamemode.c]: Good point. Kind of amazed none of the warnings or even this static analysis spotted that. Might need to add more. Will follow up with fix

Comment (mdiluz) [reply daemon/gamemode.c]: 364aea7 should sort that

**Comment by gicmo** [comment]
fwiw, the rest looks good to me ;)

**Pull request merged**